api/libproject/devstart.rb

```diff
@@ -282,12 +282,15 @@ def drop_cloud_cdr(*args)
 end
 
 def connect_to_cloud_db(*args)
+  common = Common.new
   op = WbOptionsParser.new("connect-to-cloud-db", args)
-  CloudSqlProxyContext.new(GcloudContextV2.new(op)).run do |env_file_path|
-    sleep 1 # TODO(dmohs): Detect running better.
-    env = Workbench::read_vars_file(env_file_path)
+  gcc = GcloudContextV2.new(op)
+  env = Workbench::read_vars(common.capture_stdout(%W{
```

Comment:
+1 to not saving files off if we don't have to.

At some point if you're going to transition existing targets over, you're going to want a helper method that does all this / make it run as a part of CloudSqlProxyContext or GcloudContextV2.

For now this is fine.

Reply:
Good point. For now, I like how transparent this is and I'm not sure how many times it will be used. I'll certainly rethink if I do this again somewhere else.